tests/std/tests/P0355R7_calendars_and_time_zones_zoned_time/test.cpp

```diff
@@ -0,0 +1,354 @@
+    static_assert(is_constructible_v<Zoned_seconds, string_view, Zoned_duration&> == (Result && Has_locate_zone));
+    static_assert(is_constructible_v<Zoned_seconds, TimeZonePtr, Zoned_duration&> == Result);
+    static_assert(is_constructible_v<Zoned_seconds, string_view, Zoned_duration&> == (Result && Has_locate_zone));
+}
+
+template <class TimeZonePtr, bool Has_default_zone, bool Has_locate_zone>
+constexpr void assert_constructible() {
+    using Zoned           = zoned_time<seconds, TimeZonePtr>;
+    using Zoned_no_locate = zoned_time<seconds, const Always_zero*>;
+
+    static_assert(is_constructible_v<Zoned> == Has_default_zone);
+    static_assert(is_constructible_v<Zoned, const sys_time<seconds>&> == Has_default_zone);
+    static_assert(is_constructible_v<Zoned, TimeZonePtr> == true);
+    static_assert(is_constructible_v<Zoned, string_view> == Has_locate_zone);
+    static_assert(is_constructible_v<Zoned, const Zoned&> == true);
+    static_assert(is_constructible_v<Zoned, TimeZonePtr, const sys_time<seconds>&> == true);
+    static_assert(is_constructible_v<Zoned, string_view, const sys_time<seconds>&> == Has_locate_zone);
+    static_assert(is_constructible_v<Zoned, TimeZonePtr, const local_time<seconds>&> == true);
+    static_assert(is_constructible_v<Zoned, string_view, const local_time<seconds>&> == Has_locate_zone);
+    static_assert(is_constructible_v<Zoned, TimeZonePtr, const local_time<seconds>&, choose> == true);
+    static_assert(is_constructible_v<Zoned, string_view, const local_time<seconds>&, choose> == Has_locate_zone);
+    static_assert(is_constructible_v<Zoned, TimeZonePtr, const Zoned_no_locate&> == true);
+    static_assert(is_constructible_v<Zoned, string_view, const Zoned_no_locate&> == Has_locate_zone);
+    static_assert(is_constructible_v<Zoned, TimeZonePtr, const Zoned_no_locate&, choose> == true);
+    static_assert(is_constructible_v<Zoned, string_view, const Zoned_no_locate&, choose> == Has_locate_zone);
+
+    // when (_Duration2 != _Duration)
+    assert_constructible_durations<TimeZonePtr, seconds, Has_locate_zone, true>();
+    static_assert(is_convertible_v<sys_time<hours>, sys_time<seconds>> == true);
+    assert_constructible_durations<TimeZonePtr, hours, Has_locate_zone, true>();
+    static_assert(is_convertible_v<sys_time<milliseconds>, sys_time<seconds>> == false);
+    assert_constructible_durations<TimeZonePtr, milliseconds, Has_locate_zone, false>();
+}
+
+constexpr void zonetime_constraints_test() {
+    assert_constructible<const time_zone*, true, true>();
+    assert_constructible<const Always_zero*, false, false>();
+    assert_constructible<const Has_default*, true, false>();
+    assert_constructible<const Has_locate*, false, true>();
+}
+
+void test() {
+    try {
+        zonetime_constructor_test();
+        zonetime_operator_test();
+        zonetime_exception_tests();
+        zonetime_traits_test();
+        zonetime_constraints_test();
+    } catch (exception& ex) {

```

Comment:
Should include `<exception>`. We also conventionally catch exceptions by const reference.